monai/apps/reconstruction/mri_transforms.py

```diff
@@ -0,0 +1,301 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence
+
+import numpy as np
+import torch
+from numpy import ndarray
+from torch import Tensor
+
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.utils.type_conversion import convert_data_type, convert_to_tensor
+
+
+def convert_to_tensor_complex(data: ndarray) -> Tensor:
+    """
+    Convert numpy array to PyTorch tensor.
+    For complex arrays, the real and imaginary
+    parts are stacked along the last dimension.
+
+    Args:
+        data: Input numpy array
+
+    Returns:
+        PyTorch version of the data
+
+    Example:
+        .. code-block:: python
+
+            import numpy as np
+            data = np.array([ [1+1j, 1-1j], [2+2j, 2-2j] ])
+            # the following line prints (2,2)
+            print(data.shape)
+            # the following line prints torch.Size([2, 2, 2])
+            print(convert_to_tensor_complex(data).shape)
+    """
+    if np.iscomplexobj(data):
+        data = np.stack((data.real, data.imag), axis=-1)
```

Comment:
If you use `torch.complex` does that cause problems in other places in our code? If so, where?

Reply:
Good point, no it it doesn't, but this is a problem-specific common practice. In MRI reconstruction, models are applied to real and imaginary parts as two separate channels.

Further, I believe torch.complex won't comply with PyTorch 1.7 autograd as this was featured in [the 1.8 release notes](https://pytorch.org/blog/pytorch-1.8-released/).